src/main/java/io/odpf/depot/bigquery/table/BigQueryTableClustering.java

```diff
@@ -0,0 +1,44 @@
+package io.odpf.depot.bigquery.table;
+
+import com.google.cloud.bigquery.Clustering;
+import com.google.cloud.bigquery.Field;
+import com.google.cloud.bigquery.FieldList;
+import com.google.cloud.bigquery.Schema;
+import io.odpf.depot.bigquery.exception.BQClusteringKeysException;
+import io.odpf.depot.config.BigQuerySinkConfig;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class BigQueryTableClustering {
+
+    private static final int MAX_CLUSTERING_KEYS = 4;
+    private final BigQuerySinkConfig bqConfig;
+
+
+    public BigQueryTableClustering(BigQuerySinkConfig bqConfig) {
+        this.bqConfig = bqConfig;
+    }
+
+    public Clustering getClusteredTableDefinition(Schema schema) {
```

Comment:
ah. we don't need these many classes. These are static methods. These classes are providing anything.
we should go back to the structure that we had before. And have partitioning and clustering config in the same class.